python/pipeline/pupil.py

```diff
@@ -540,4 +575,998 @@ def make(self, key):
 
         self.insert1(key)
         for ckey in tqdm(contours):
-            self.Ellipse().insert1(ckey, ignore_extra_fields=True)
\ No newline at end of file
+            self.Ellipse().insert1(ckey, ignore_extra_fields=True)
+
+# If config.yaml ever updated, make sure you store the file name differently so that it becomes unique
+@schema
+class ConfigDeeplabcut(dj.Manual):
+    definition = """
+    # Minimal info needed to load deeplabcut model
+    config_path         : varchar(255)          # path to deeplabcut config
+    ---
+    shuffle             : smallint unsigned     # shuffle number used for the trained dlc model. Needed for dlc.analyze_videos
+    trainingsetindex    : smallint unsigned     # trainingset index used for the trained dlc. model. Needed for dlc.analyze_videos
+    """
+
+@schema
+class TrackingMethod(dj.Manual):
+    definition="""
+    -> Eye
+    tracking_method     : tinyint unsigned # tracking_method 0:manual 1: deeplabcut
+    ---
+    """
+
+    def fill(self, key, tracking_method):
+        self.insert1(dict(key, tracking_method=tracking_method))
+
+@schema
+class Tracking(dj.Computed):
+    definition="""
+    -> TrackingMethod
+    ---
+    tracking_ts=CURRENT_TIMESTAMP   : timestamp  # automatic
+    """
+
+    class ManualTracking(dj.Part):
+        definition="""
+        -> master
+        frame_id                    : int                   # frame id with matlab based 1 indexing
+        ---
+        contour=NULL                : longblob              # eye contour relative to ROI
+        """
+        def make(self, key):
+            print("""
```

Comment:
what I was thinking is to move the make of ManuallyTrackedContours here and put the sign about deprecation in ManuallyTrackedContours, that way all tracking code is in the same place and if you wanna track with deeplabcut or manually you would use the same table (just with a different tracking method). So move all code from ManuallyTrackedContours here, and if somebody tries to populate that table just tell them to populate this with tracking_method: 0 (assuming 0 is manual method); there's only three people (jiakun, taliah and paul) that manually track stuff (and lately it's been only Taliah) so it won't be a problem to ask em to do it this way now. We should also have this function but call it something like migrate (so whatever was in the old ManuallyTracked could be migrated this function)

Reply:
You also need to standardize the saving of results between the two methods. Anything that depends on Tracking shouldn't care how stuff is tracked, it should be able to just read whatever is tracked otherwise there is no use to have them in the same table (for instance we have methods to do diff segmentations or deconvolutions or registrations in the same table but they all save the same results in a shared table, that could be a part table; that way from there on people don't need to care how stuff was segmented or deconvolved, etc.). I was assuming that at some point you transform manual data or deeplabcut data into some shared, standard format that you then fit the circle and ellipse to, this may not be true (should it?), if it isn't I don't see why we are putting it in the same table. It seems like the output of tracking should be the same in both cases, but if it's not they should remain in two distinct tables (because they are not functionally related); the output of fitting will be the same (again, I guess) so maybe there is where we should join stuff

Reply:
I thought we initially were gonna make the joining in the fitted something like:
```python
class ManuallyTrackedContours() # or whatever name it already has
-> Eye
---
manual_tracked _results

class DeepLabCutTrack() 
->Eye
---
deeplabcuttracked_results

class NewFittedContour():
-> Eye
-> shared.TrackingMethod
---
fitted_results
  class ManuallyTrackedFitting(Part):
    definition = """
    -> ManualTrackedContours
    """
    def make():
      # make here will do the fitting bringing data from MAnuallyTrackedContours and inserts the results in NEwFittedContour (
  class DeepLabCutFitted(dj.Part):
    definition = """
    ->DeepLabCutTrack
    """"
    def make():
      # make here again brings stuff from  DeepLabCutTrack and does the fitting
  def make():
    if tracking_method=1:
      self.insert1(key)
      DeepLabcUtftting.make()
    if tracknig_method 2:
      self.insert1(key)
      ManualFitting.make()
```

Reply:
we'll leave as is